tsfc/fem.py

```diff
@@ -233,6 +234,11 @@ def flat_index(ordered_deriv):
         return result[0]
 
 
+def zero_tracking_literal(table, ufl_element):
+    track_zeros = isinstance(ufl_element, (ufl.EnrichedElement, ufl.MixedElement))
```

Comment:
Correct. As I said, the idea, perhaps naive at that time I don't know, was to exploit the "knowledge" that in non-scalar function spaces we have "non-degenerate" blocks of zero-valued entries (non-degenerate ---> not blocks of size Nx1, i.e. spare columns)